crates/aptos/src/move_tool/fmt.rs

```diff
@@ -0,0 +1,147 @@
+}
+
+#[derive(Debug, Args)]
+pub struct FmtCommand {
+    /// How to generate and show the result after reformatting
+    #[clap(long, value_enum, default_value = "overwrite")]
+    emit_mode: EmitMode,
+
+    /// Path to the file to be formatted
+    #[clap(long)]
+    file_path: Option<PathBuf>,
+
+    /// Path to the directory to be formatted
+    /// if neither file_path or dir_path is set, all move files in the current folder will be formatted
+    #[clap(long)]
+    dir_path: Option<PathBuf>,
+
+    /// Path for the configuration file
+    /// Recursively searches the given path for the
+    /// movefmt.toml config file
+    #[clap(long, value_parser)]
+    pub config_path: Option<PathBuf>,
+
+    /// Set options from command line. These settings take
+    /// priority over .movefmt.toml.
+    /// Config options can be found at https://github.com/movebit/movefmt/blob/develop/doc/how_to_use.md
+    #[clap(long, value_parser = crate::common::utils::parse_map::<String, String>, default_value = "")]
+    pub(crate) config: BTreeMap<String, String>,
+
+    #[clap(long, short)]
+    /// Print verbose output
+    pub verbose: bool,
+
+    #[clap(long, short)]
+    /// Print less output
+    pub quiet: bool,
+}
+
+#[async_trait]
+impl CliCommand<String> for Fmt {
+    fn command_name(&self) -> &'static str {
+        "Fmt"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<String> {
+        self.command.execute().await
+    }
+}
+
+impl FmtCommand {
+    async fn execute(self) -> CliTypedResult<String> {
+        let exe = get_movefmt_path()?;
+        let mut cmd = Command::new(exe.as_path());
+        let input_opt = self.file_path;
+        let dir_opt = self.dir_path;
+        let config_path_opt = self.config_path;
+        let config_map = self.config;
+        let verbose_flag = self.verbose;
+        let quiet_flag = self.quiet;
+        let emit_mode = match self.emit_mode {
+            EmitMode::Overwrite => "overwrite",
+            EmitMode::NewFile => "new_file",
+            EmitMode::StdOut => "stdout",
+            EmitMode::Diff => "diff",
+        };
+        cmd.arg(format!("--emit={}", emit_mode));
+        if let Some(config_path) = config_path_opt {
+            cmd.arg(format!("--config-path={}", config_path.as_path().display()));
+        }
+        if verbose_flag {
+            cmd.arg("-v");
+        } else if quiet_flag {
+            cmd.arg("-q");
+        }
+        if !config_map.is_empty() {
+            let mut config_map_str = "".to_string();
+            let mut first = true;
+            for (key, value) in config_map {
+                if first {
+                    config_map_str = format!("{}={}", key, value);
+                    first = false;
+                } else {
+                    config_map_str = format!("{},{}={}", config_map_str, key, value);
+                }
+            }
+            cmd.arg(format!("--config={}", config_map_str));
+        }
+        if input_opt.is_some() {
```

Comment:
You can should probably make input and dir mutually exclusive at the clap level, vs now where dir opt gets silently ignored if input opt is set.

Reply:
done